src/exactio/exactio_exanic.c

```diff
@@ -231,15 +231,24 @@ static int set_exanic_params(exanic_t *exanic, char* device, int port_number,
         exit(1);
     }
 
+    int ifr_changed = 0;
+
     if(promisc)
+    {
+        ifr_changed |= !(ifr.ifr_flags & IFF_PROMISC);
         ifr.ifr_flags |= IFF_PROMISC;
+    }
 
+    ifr_changed |= !(ifr.ifr_flags & IFF_UP);
     ifr.ifr_flags |= IFF_UP;
 
-    if (ioctl(fd, SIOCSIFFLAGS, &ifr) == -1)
+    if (ifr_changed)
     {
-        fprintf(stderr, "ioctl(SIOCSIFFLAGS): %s:%d: %s\n", device, port_number, strerror(errno));
-        exit(1);
+        if (ioctl(fd, SIOCSIFFLAGS, &ifr) == -1)
+        {
+            fprintf(stderr, "ioctl(SIOCSIFFLAGS): %s:%d: %s\n", device, port_number, strerror(errno));
```

Comment:
I really like that you've made these errors more explicit, but can I ask for two quick changes before I merge in:
1) can you please replace fprintf() with the libchaste logging functions ch_log_XXX which are used throughout the project? I'd suggest ch_log_error() or ch_log_fatal() in this instance. See https://github.com/exablaze-oss/exact-capture/blob/master/libs/libchaste/include/log/log.h
2) Can you make the error text more descriptive? See https://github.com/exablaze-oss/exact-capture/blob/89615ae0f9ceed8a7ef5df78a1189dec32cb924b/src/exact-capture-listener.c#L449

Thanks!